app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java

```diff
@@ -0,0 +1,57 @@
+package com.github.sgueissa.bootcamp;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.concurrent.CompletableFuture;
+
+public class FirebaseBootcamp extends AppCompatActivity {
+
+    private DatabaseReference db = FirebaseDatabase.getInstance().getReference();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_firebase_bootcamp);
+    }
+
+    public void get(View view){
+        TextView email = (TextView) findViewById(R.id.email_text);
+        TextView phone = (TextView) findViewById(R.id.phone_text);
+        CompletableFuture<String> future = new CompletableFuture<>();
+        db.child(phone.getText().toString()).get().addOnSuccessListener(new OnSuccessListener<DataSnapshot>() {
+            @Override
+            public void onSuccess(DataSnapshot dataSnapshot) {
+                if (dataSnapshot.getValue() == null) future.completeExceptionally(new NoSuchFieldException());
+                else future.complete(dataSnapshot.getValue().toString());
+            }
+        }).addOnFailureListener(
+                new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        future.completeExceptionally(e);
+                    }
+                }
+        );
```

Comment:
Maybe we could use lambda expressions here to make it more concise.

Reply:
Addressed in commit [43f7f86]